docs/cpp_docs.rst

```diff
@@ -26,94 +44,556 @@ The beginning of both header and source code files should include
+
+Parameters should be listed in the same order as they appear in the class, function, or method signature. 
+
+.. code-block:: cpp
+
+   /**
+    * Storage for arbitrary data with log(N) lookup.
     *
-    * @brief This message displayed in Doxygen Files index
+    * ...
     *
-    * @ingroup PackageName
-    * (Note: this needs exactly one @defgroup somewhere)
+    * @tparam T the type of data stored in the table
+    * @tparam ComparatorT callable defining a strict weak ordering for objects
+    *     of type `T`. Its `operator()` must accept two `T` and return `true`
+    *     if and only if the first argument comes before the second. It must
+    *     not throw exceptions.
+    */
+   template <typename T, typename ComparatorT = std::less<T>>
+   class LookupTable
+   {
+       ...
+   }
+
+When two or more consecutive template parameters have *exactly* the same description, they can be combined:
+
+.. code-block:: cpp
+
+   /**
+    * @param T,U the types of the pair components
+    */
+
+Do not use a space between the parameters.
+
+.. _cpp-doxygen-parameters:
+
+Function/Method Parameters
+--------------------------
+
+A series of ``@param`` tags, usually one for each parameter. Each tag should have a description following the parameter name. You do *not* usually need to document default arguments; Doxygen will provide the default automatically. If the description extends over multiple lines, each line after the first must be indented.
+
+Parameters should be listed in the same order as they appear in the function or method signature.
+
+``@param`` should be given with the ``[in]``, ``[out]``, or ``[in,out]`` tag if the function method contains any output parameters. The ``[in]`` tag is optional if all parameters are input, even if other functions or methods in the same class or package use output parameters.
+
+.. code-block:: cpp
+
+   /**
+    * Compute mean and standard deviation for a collection of data.
     *
-    * @author Joe Smith
-    * Contact: js@lsst.org
+    * @param[out] mean the mean of `data`, or `NaN` if `data` is empty
+    * @param[out] stdDev the unbiased (sample) standard deviation, or `NaN`
+    *             if `data` contains fewer than 2 elements
+    * @param[in] data the data to analyze
+    */
+   std::ostream operator<<(double & mean, double & stdDev, std::vector<double> const & data);
+
+When two or more consecutive parameters have *exactly* the same description, they can be combined:
+
+.. code-block:: cpp
+
+   /**
+    * @param x,y the coordinates where the function is evaluated
+    */
+
+Do not use a space between the parameters.
+
+.. _cpp-doxygen-parameters-inline:
+
+Annotating Parameters with Inline Comments (optional)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If the parameter descriptions are very short, you may choose to annotate parameters with inline comments after each parameter, one per line.
+These comments are prefixed with ``///<``.
```

Comment:
Could we get away with saying "For historical reasons, this format is permitted but not encouraged, and should not be used in new code."? There's some wording like that in the main style guide...

@r-owen I just added an [example of a documented constant](https://developer.lsst.io/v/DM-7891/docs/cpp_docs.html#documenting-constants-variables-and-data-members) using the `/* ... */` style rather than `///` or `///<`. I admit the latter would save some space...